src/collection/account.service.ts

```diff
@@ -354,25 +354,40 @@ export class AccountService {
           );
         const projectName = projectDtoArr[0].collection.name;
         try {
-          this.auditService.recordToolsSync(
-            'start',
-            'unknown',
-            `Start sync: ${service.collection.scmUrl}`,
-            projectName,
-            serviceName,
-          );
-          await this.githubService.refresh(
-            projectName,
-            serviceName,
-            service.collection.scmUrl,
-          );
-          this.auditService.recordToolsSync(
-            'end',
-            'success',
-            `End sync: ${service.collection.scmUrl}`,
-            projectName,
-            serviceName,
-          );
+          const githubRegex = /^https:\/\/github\.com[:/](.+?)\/(.+?)(\.git)?$/;
+          if (
+            service.collection.scmUrl &&
+            service.collection.scmUrl.trim() !== '' &&
+            githubRegex.test(service.collection.scmUrl)
+          ) {
+            this.auditService.recordToolsSync(
+              'start',
+              'unknown',
+              `Start sync: ${service.collection.scmUrl}`,
+              projectName,
+              serviceName,
+            );
+            await this.githubService.refresh(
+              projectName,
+              serviceName,
+              service.collection.scmUrl,
+            );
+            this.auditService.recordToolsSync(
+              'end',
+              'success',
+              `End sync: ${service.collection.scmUrl}`,
+              projectName,
+              serviceName,
+            );
```

Comment:
Will still throw on "valid" permission errors and block subsequent updates. In this case, throws should be collected into an array and a aggregate error (or just the first) thrown outside of the loop.